linux/README.md

```diff
@@ -0,0 +1,107 @@
+# The Flutter Linux Desktop Embedder
+
+This framework provides the basic functionality for embedding a Flutter app on
+the Linux desktop. This currently includes support for:
+
+*  Drawing a Flutter view.
+*  Mouse event handling.
+
+# How to Use This Code
+
+Note that the example build here is `host_debug_unopt` (you'll find more info on
+what that means in this section). This can be changed to whatever version of
+the Flutter Engine you end up building.
+
+First you will need to install the relevant dependencies.
+
+## Dependencies
+
+Requires GLFW3 on your system (example for debian-based systems):
+
+```
+$ sudo apt-get install libglfw3-dev
+```
+
+Next you will need to follow the steps outlined on the flutter engine's
+[contributing](https://github.com/flutter/engine/blob/master/CONTRIBUTING.md)
+document so that you can build a copy of `libflutter_engine.so`.
+
+Then copy it in the `linux/library/` directory.
+
+```
+$ cp <path_to_flutter_engine>/src/out/host_debug_unopt/libflutter_engine.so \
+       <path_to_flutter_desktop_embedding_repo>/linux/library
+```
+
+Then copy `embedder.h` into the include directory (the file can be found on [this
+page](https://github.com/flutter/engine/wiki/Custom-Flutter-Engine-Embedders),
+under [this
+link](https://github.com/flutter/engine/blob/4733e3373789894aa4f593137c6d440891d492a2/shell/platform/embedder/embedder.h), or you can copy it from the flutter engine checkout)
+
+```
+$ cp <path_to_flutter_engine>/src/flutter/shell/platform/embedder/embedder.h \
+     <path_to_flutter_desktop_embedding_repo>/linux/library/include
+```
+
+After that's all done you should be able to run `make` within the `linux`
+directory and have access to a `flutter_embedder_example` binary.
+
+```
+$ make
+```
+
+
+## Building and Running a Flutter App
+
+
+There are already [examples](https://flutter.io/get-started/) on building
+Flutter apps out there. This assumes you've already got an example somewhere.
+
+For this you'll need a working `flutter` binary. An easy way to get both the
+`flutter` binary and an example app is to clone the [flutter
+repo](https://github.com/flutter/flutter), and you can then build, for example,
+the demo gallery.
```

Comment:
This repo has an example app in it, which is what I'd like all the platform examples to use. That way if we want to add specific feature examples (e.g., a plugin with Dart-side code) it'll apply to all platforms, and we can ensure that it has all the other features we want to be able to easily test over time (like adding a text field).

Reply:
Updated